### ISSUE TITLE
调制类型识别

### ISSUE THREAD

**Issue by littletree123** [framist/3rd-EBDSC#1]
您好，我想只做调制类型识别任务，请问要怎么做呢

**Comment by framist** [comment]
https://github.com/framist/3rd-EBDSC/blob/9c8ad2bcdf8c19406d166da598b02e35e6e61105/ModernTCN_FreTS.py#L479-L485

https://github.com/framist/3rd-EBDSC/blob/9c8ad2bcdf8c19406d166da598b02e35e6e61105/tcn_3rd.py#L398-L409

修剪类似此处的分类头与相关逻辑即可

**Comment by littletree123** [comment]
> [3rd-EBDSC/ModernTCN_FreTS.py](https://github.com/framist/3rd-EBDSC/blob/9c8ad2bcdf8c19406d166da598b02e35e6e61105/ModernTCN_FreTS.py#L479-L485)
> 
> Lines 479 to 485 in [9c8ad2b](/framist/3rd-EBDSC/commit/9c8ad2bcdf8c19406d166da598b02e35e6e61105)
> 
>  # = 调制类型分类 
>  mod_logits = self.mod_classifier(global_feat)  # [batch_size, num_mod_classes] 
>   
>  # = 码元宽度回归 
>  symbol_width = self.symbol_width_regressor(global_feat).squeeze(-1)  # [batch_size] 
>   
>  return mod_logits, symbol_width, code_seq_logits 
> [3rd-EBDSC/tcn_3rd.py](https://github.com/framist/3rd-EBDSC/blob/9c8ad2bcdf8c19406d166da598b02e35e6e61105/tcn_3rd.py#L398-L409)
> 
> Lines 398 to 409 in [9c8ad2b](/framist/3rd-EBDSC/commit/9c8ad2bcdf8c19406d166da598b02e35e6e61105)
> 
>  mod_logits, symbol_width_pred, code_seq_logits = model(IQ_data) 
>   
>  # 计算损失 
>  loss = criterion( 
>      mod_logits=mod_logits, 
>      symbol_width_pred=symbol_width_pred, 
>      code_seq_logits=code_seq_logits, 
>      mod_labels=mod_type, 
>      symbol_width_labels=symbol_width, 
>      code_seq_labels=code_sequence_aligned, 
>      code_seq_masks=code_mask, 
>  ) 
> 修剪类似此处的分类头与相关逻辑即可

谢谢！
